ops/framework.py

```diff
@@ -878,7 +882,24 @@ def _reemit(self, single_event_path: str = None):
             observer = self._observer.get(observer_path)
             if observer:
                 if single_event_path is None:
-                    logger.debug("Re-emitting %s.", event)
+                    logger.debug("Re-emitting deferred %s.", event)
+                elif isinstance(event, FrameworkEvent):
+                    pass  # We get two of these each hook execution: too much noise.
+                else:
+                    try:
```

Comment:
The argument here is likely for something which is _not_ being fired by Juju, or a hook, or even an action. But something like:

```python
class SomeCharmLibraryEvent(RelationEvent):
  """Emit an event with a relation property for some workflow."""
  
class SomeInterfaceProviderEvents(ObjectEvents):
  some_event = EventSource(SomeCharmLibraryEvent)
...

```

This has become a reasonable-enough pattern to decouple libraries and "actual" charm code, rather than any library trying to call charm methods/attributes directly.

Re-parenting `CommitEvent` as a `FrameworkEvent` is an attempt to ensure that serializing data (to an event for dispatch, for example) doesn't emit a 1-2 punch for logging. Passing this up from `_Dispatcher` would definitely be nicer than reading env vars here, agreed.

Reply:
Like Barry said. 
I was searching for a nicer alternative to reading up the envvar, and this sounds good. I'll pass it up from _Dispatcher.